python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -275,6 +275,44 @@ def __iter__(self):
         for i, name in enumerate(group_names):
             yield name, grouped_values[offsets[i] : offsets[i + 1]]
 
+    @property
+    def dtypes(self):
+        """
+        Return the dtypes in this group.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The data type of each column of the group.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> df = cudf.DataFrame({'a': [1, 2, 3, 3], 'b': ['x', 'y', 'z', 'a'],
+        ...                      'c':[10, 11, 12, 12]})
+        >>> df.groupby("a").dtypes
+                b      c
+        a
+        1  object  int64
+        2  object  int64
+        3  object  int64
+        """
+        non_grouped = [
+            name
+            for name in self.obj._data.names
+            if name not in self.grouping.names
+        ]
+        all_dtypes = self.obj._dtypes
+        index = self.grouping.keys.unique().to_pandas()
+        df = pd.DataFrame(
+            {
+                name: pd.Series([all_dtypes[name]]).repeat(len(index))
+                for name in non_grouped
+            }
+        )
+        df.index = index
+        return df
```

Comment:
Can make this more concise:
```suggestion
        return pd.DataFrame(
            {
                name: pd.Series([self.obj._dtypes[name]]).repeat(len(index))
                for name in non_grouped
            },
            index=self.grouping.keys.unique().to_pandas()
        )
```

Reply:
@vyasr I had to revert part of your code suggestion because we cannot directly pass `index` here to the DataFrame constructor as pandas is force re-indexing all the Series objects with the `index`, leading to mismatching values because of mismatching indexes.